examples/training/object_detection/pascal_voc/retina_net.py

```diff
@@ -75,68 +76,241 @@
+            height > width,
+            lambda: float(IMG_SIZE),
+            lambda: tf.cast(height / width * IMG_SIZE, tf.float32),
+        )
+        target_width = tf.cond(
+            width > height,
+            lambda: float(IMG_SIZE),
+            lambda: tf.cast(width / height * IMG_SIZE, tf.float32),
+        )
+        image = tf.image.resize(
+            raw_image, (target_height, target_width), antialias=False
+        )
+
+        gt_boxes = keras_cv.bounding_box.convert_format(
+            inputs["objects"]["bbox"],
+            images=image,
+            source="rel_yxyx",
+            target="xyxy",
+        )
+        image = tf.image.pad_to_bounding_box(
+            image, 0, 0, target_size[0], target_size[1]
+        )
+        gt_boxes = keras_cv.bounding_box.convert_format(
+            gt_boxes,
+            images=image,
+            source="xyxy",
             target=bounding_box_format,
         )
+        gt_classes = tf.cast(inputs["objects"]["label"], tf.float32)
+        gt_classes = tf.expand_dims(gt_classes, axis=-1)
+
         bounding_boxes = tf.concat([gt_boxes, gt_classes], axis=-1)
-        return {"images": image, "bounding_boxes": bounding_boxes}
+
+        return image, bounding_boxes
 
     return apply
 
 
-train_ds = train_ds.map(unpackage_inputs("xywh"), num_parallel_calls=tf.data.AUTOTUNE)
+def pad_fn(images, boxes):
+    boxes = boxes.to_tensor(default_value=-1.0, shape=[GLOBAL_BATCH_SIZE, 32, 5])
+    gt_boxes = boxes[:, :, :4]
+    gt_classes = boxes[:, :, 4]
+    return images, {"boxes": gt_boxes, "classes": gt_classes}
+
+
+train_ds = train_ds.map(
+    proc_train_fn("xywh", image_size), num_parallel_calls=tf.data.AUTOTUNE
+)
+
 train_ds = train_ds.apply(
     tf.data.experimental.dense_to_ragged_batch(GLOBAL_BATCH_SIZE, drop_remainder=True)
 )
-
+train_ds = train_ds.map(pad_fn, num_parallel_calls=tf.data.AUTOTUNE)
 train_ds = train_ds.shuffle(8 * strategy.num_replicas_in_sync)
 train_ds = train_ds.prefetch(tf.data.AUTOTUNE)
 
 eval_ds = eval_ds.map(
-    unpackage_inputs("xywh"),
+    proc_eval_fn(bounding_box_format="xywh", target_size=image_size),
     num_parallel_calls=tf.data.AUTOTUNE,
 )
 eval_ds = eval_ds.apply(
     tf.data.experimental.dense_to_ragged_batch(GLOBAL_BATCH_SIZE, drop_remainder=True)
 )
+eval_ds = eval_ds.map(pad_fn, num_parallel_calls=tf.data.AUTOTUNE)
 eval_ds = eval_ds.prefetch(tf.data.AUTOTUNE)
 
-
-"""
-Our data pipeline is now complete.  We can now move on to data augmentation:
-"""
-eval_resizing = layers.Resizing(
-    IMG_SIZE, IMG_SIZE, bounding_box_format="xywh", pad_to_aspect_ratio=True
-)
-
-augmenter = layers.Augmenter(
-    [
-        layers.RandomFlip(mode="horizontal", bounding_box_format="xywh"),
-        layers.JitteredResize(
-            target_size=(IMG_SIZE, IMG_SIZE),
-            scale_factor=(0.8, 1.25),
-            bounding_box_format="xywh",
-        ),
-        layers.MaybeApply(layers.MixUp(), rate=0.5, batchwise=True),
```

Comment:
What if we just remove the MixUp KPL?
(Since the flip and resize are at least theoretically the same in KPL and non-KPL form, I figure it's worth a shot)

Reply:
I tried, slightly better, but still worse than without KPL